src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,185 @@
+import urllib.request
+import json
+import tempfile
+import platform
+import os
+import subprocess
+import winreg
+import ctypes
+import sys
+
+# <TODO> repelace the URL with a static URL
+_META_DATA_URL = "http://localhost:8000/ni-daqmx-update-metadata.json"
+
+
+def _is_admin():
+    """
+    Verify whether the user has Administrator privileges.
+    """
+    try:
+        return ctypes.windll.shell32.IsUserAnAdmin()
+    except Exception as e:
+        print(f"Error checking admin privileges: {e}")
+        return False
+
+
+def _parse_version(version):
+    """
+    Split the version string into an array of integers.
+    """
+    return [int(part) for part in version.split(".")]
+
+
+def _compare_version(installed_ver, source_ver):
+    """
+    Compare two versions which are in the format: major.minor.update
+    """
+    iver_parts = _parse_version(installed_ver)
+    sver_parts = _parse_version(source_ver)
+
+    for iver_num, sver_num in zip(iver_parts + [0], sver_parts + [0]):
+        if iver_num < sver_num:
+            return -1
+        elif iver_num > sver_num:
+            return 1
+
+    return 0
+
+
+def _check_daqmx_is_installed():
+    """
+    Check for existing installation of DAQmx by checking for Version information.
+    Version information is read from the registry key.
+    """
+    try:
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            return product_version
+    except Exception as e:
```

Comment:
This exception should probably be more specific.
- `winreg` should raise `OSError`, you should catch that.
- `OSError` indicates the Windows error code. If you check the error code, you can distinguish between key-not-found and other errors. That way, you don't treat other errors as "DAQmx is not installed".